src/module.c

```diff
@@ -7930,22 +7930,29 @@ int RM_InfoEndDictField(RedisModuleInfoCtx *ctx);
  * NULL or empty string indicates the default section (only `<modulename>`) is used.
  * When return value is REDISMODULE_ERR, the section should and will be skipped. */
 int RM_InfoAddSection(RedisModuleInfoCtx *ctx, char *name) {
+
     sds full_name = sdsdup(ctx->module->name);
-    if (name != NULL && strlen(name) > 0)
+    sds lower_full_name = sdsdup(ctx->module->name);
+    if (name != NULL && strlen(name) > 0){
         full_name = sdscatfmt(full_name, "_%s", name);
-
+        lower_full_name = sdscatfmt(lower_full_name, "_%s", name);        
+    }
+        
     /* Implicitly end dicts, instead of returning an error which is likely un checked. */
     if (ctx->in_dict_field)
         RM_InfoEndDictField(ctx);
 
+    sdstolower(lower_full_name);
+
     /* proceed only if:
      * 1) no section was requested (emit all)
      * 2) the module name was requested (emit all)
      * 3) this specific section was requested. */
-    if (ctx->requested_section) {
-        if (strcasecmp(ctx->requested_section, full_name) &&
-            strcasecmp(ctx->requested_section, ctx->module->name)) {
+    if (ctx->requested_sections) {
+        if ((dictFind(ctx->requested_sections,lower_full_name) == NULL) &&
+            (dictFind(ctx->requested_sections,ctx->module->name) == NULL)) {
```

Comment:
Here, I use sdstolower(full_name) lower module name.

Reply:
yes, but you only lowered the `full_name` (module + section).
the other part of the `if` is on `ctx->module->name` which should also be lowered.

unless we'll use a case-insensitive dict hash + compare (e.g. dictGenCaseHashFunction)

Reply:
Sorry, Oran. Here I am a little bit confused why you think ctx->module->name should be lowered as well. 
When I do test, I do not find the reason and I am not familiar with the Module part. Could you please give me one example on this. Thanks.

Reply:
RM_CreateDataType says
```
 * * **name**: A 9 characters data type name that MUST be unique in the Redis
 *   Modules ecosystem. Be creative... and there will be no collisions. Use
 *   the charset A-Z a-z 9-0, plus the two "-_" characters. A good
 *   idea is to use, for example `<typename>-<vendor>`. For example
 *   "tree-AntZ" may mean "Tree data structure by @antirez". To use both
 *   lower case and upper case letters helps in order to prevent collisions.
```
so a module name could have upper case characters, and in INFO we wanna do case-insensitive matching for the section names, so the module section (in case the module didn't explicitly define a section name) could be uppercase.

Reply:
Since the module name is case sensitive, there appears to be a risk that a case-insensitive search on the module name could generate false matches.  Is that acceptable behavior?

Reply:
you're right. but since info sections are case-insensitive, i think i'd rather return two sections in case there are two modules with similar name, rather than not return a section because the INFO command was used with non-exact case.